streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -440,7 +466,6 @@ public synchronized void start() throws IllegalStateException, StreamsException
 
         if (state == State.CREATED) {
             checkBrokerVersionCompatibility();
-            setState(State.RUNNING);
```

Comment:
Good catch. I'm hesitant to add another state since this is user facing and all the user wants to know is if we're running. I'd propose going back to what we had. The original problem was that that'd lead to a RUNNING->RUNNING transition, but I can now check for that and ignore if that's the case. 

@guozhangwang the `tryTransitTo` function does not solve this particular problem with avoiding starting twice unfortunately.

Also what is the difference of the `tryTransitTo` and the current `setState` that has similar logic?